repos/system_upgrade/el7toel8/actors/openldapnsstoopenssl/libraries/nss.py

```diff
@@ -0,0 +1,108 @@
+import os
+import os.path as ph
+import re
+import subprocess as sp
```

Comment:
```suggestion
from leapp.libraries.stdlib import run
```
Please use _Leapp_ library to run external commands, it makes easier for debugging and also uses verbosity arguments for the outputs.

Reply:
Nice catch. The subprocess library is forbidden to use in actors.

Reply:
I've fixed most of these in the latest commit. However, in the one (two) last case I'm actually doing a pipe (Popen.stdout -> Popen.stdin -> .stdout -> fd.write). For this case I have not found any feasible workaround since the `run` procedure does not provide an option to provide an stdin stream. Any suggestion other than an intermediate temporary file (which is fairly "ugly" since a private key is in the game)?

Reply:
Hmm, the lower level `_call` function allows to pass in `stdin` as either string/numeric file descriptor. I've created https://github.com/oamg/leapp/issues/579 to track this . That being said, I think this is OK, especially because it handles secrets.